crates/pop-cli/src/common/mod.rs

```diff
@@ -16,3 +16,219 @@ pub mod prompt;
 /// Contains runtime utilities.
 pub mod runtime;
 pub mod wallet;
+
+use std::fmt::{Display, Formatter, Result};
+use strum::VariantArray;
+
+#[derive(Debug, PartialEq)]
+pub enum Telemetry {
+	Null,
+	Build(Project),
+	Test { project: Project, feature: Feature },
+	Install(Os),
+	Up(Project),
+	New(Template),
+}
+
+/// Represents the type of project being operated on.
+#[derive(Debug, PartialEq, Clone, VariantArray)]
+pub enum Project {
+	Network,
```

Comment:
nitpick: missing doc comment. Will just leave this one comment as there are other pubs missing doc comments and this is a high-level review (as requested)